azurerm/resource_arm_dev_test_lab_schedule.go

```diff
@@ -0,0 +1,489 @@
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmDevTestLabSchedules() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmDevTestLabSchedulesCreateUpdate,
+		Read:   resourceArmDevTestLabSchedulesRead,
+		Update: resourceArmDevTestLabSchedulesCreateUpdate,
+		Delete: resourceArmDevTestLabSchedulesDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": azure.SchemaLocation(),
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"dev_test_lab_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"status": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  dtl.EnableStatusDisabled,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(dtl.EnableStatusEnabled),
+					string(dtl.EnableStatusDisabled),
+				}, true),
+			},
+
+			"task_type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"weekly_recurrence": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"time": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+
+						"week_days": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+								ValidateFunc: validation.StringInSlice([]string{
+									string(scheduler.Monday),
+									string(scheduler.Tuesday),
+									string(scheduler.Wednesday),
+									string(scheduler.Thursday),
+									string(scheduler.Friday),
+									string(scheduler.Saturday),
+									string(scheduler.Sunday),
+								}, true),
+								DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+							},
+							// Set: set.HashStringIgnoreCase,
+						},
+					},
+				},
+			},
+
+			"daily_recurrence": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"time": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			"hourly_recurrence": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"minute": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			"time_zone_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"notification_settings": {
+				Type:     schema.TypeSet,
```

Comment:
since there's only a single element available here can we make this a TypeList instead:

```suggestion
				Type:     schema.TypeList,
```

Reply:
Done